packages/story-editor/src/app/api/apiProvider.js

```diff
@@ -173,369 +112,141 @@ function APIProvider({ children }) {
-        page,
-        _web_stories_envelope: true,
-      });
-      return apiFetch({ path: apiPath }).then(({ headers, body }) => {
-        const totalPages = parseInt(headers['X-WP-TotalPages']);
-        const templates = body.map((template) => {
-          return { ...template['story_data'], templateId: template.id };
-        });
-        return {
-          templates,
-          hasMore: totalPages > page,
-        };
-      });
-    },
-    [customPageTemplates]
+  // @todo Move to wp-story-editor along with meta-boxes.
+  actions.saveMetaBoxes = useCallback(
+    (story, formData) => saveMetaBoxes(story, formData, metaBoxes),
+    [metaBoxes, saveMetaBoxes]
   );
 
-  const addPageTemplate = useCallback(
-    (page) => {
-      return apiFetch({
-        path: `${customPageTemplates}/`,
-        data: {
-          story_data: page,
-          status: 'publish',
-        },
-        method: 'POST',
-      }).then((response) => {
-        return { ...response['story_data'], templateId: response.id };
-      });
-    },
-    [customPageTemplates]
+  // @todo Move to wp-story-editor along with StatusCheck component.
+  actions.getStatusCheck = useCallback(
+    (content) => getStatusCheck(content, statusCheck, encodeMarkup),
+    [statusCheck, encodeMarkup, getStatusCheck]
   );
 
-  const deletePageTemplate = useCallback(
-    (id) => {
-      // `?_method=DELETE` is an alternative solution to override the request method.
-      // See https://developer.wordpress.org/rest-api/using-the-rest-api/global-parameters/#_method-or-x-http-method-override-header
-      return apiFetch({
-        path: addQueryArgs(`${customPageTemplates}${id}/`, {
-          _method: 'DELETE',
-        }),
-        data: { force: true },
-        method: 'POST',
-      });
-    },
-    [customPageTemplates]
+  actions.getCustomPageTemplates = useCallback(
+    (page = 1) => getCustomPageTemplates(page, customPageTemplates),
+    [customPageTemplates, getCustomPageTemplates]
   );
 
-  const state = {
-    actions: {
-      autoSaveById,
-      getStoryById,
-      getDemoStoryById,
-      getStoryLockById,
-      setStoryLockById,
-      deleteStoryLockById,
-      getMedia,
-      getHotlinkInfo,
-      getLinkMetadata,
-      saveStoryById,
-      getAuthors,
-      uploadMedia,
-      updateMedia,
-      deleteMedia,
-      saveMetaBoxes,
-      getStatusCheck,
-      addPageTemplate,
-      getCustomPageTemplates,
-      deletePageTemplate,
-      getPageTemplates,
-      getCurrentUser,
-      updateCurrentUser,
-    },
-  };
+  actions.addPageTemplate = useCallback(
+    (page) => addPageTemplate(page, customPageTemplates),
+    [customPageTemplates, addPageTemplate]
+  );
+
+  actions.deletePageTemplate = useCallback(
+    (id) => deletePageTemplate(id, customPageTemplates),
+    [customPageTemplates, deletePageTemplate]
+  );
+
+  Object.entries(apiCallbacks).forEach(([name, callback]) => {
+    if (!callback) {
+      actions[name] = undefined;
```

Comment:
Alternative:

```js
actions[name] = () => {};
```

WDYT?

Reply:
It would be easier to check `undefined`, than to check an empty function.

Reply:
If it's an empty function we wouldn't have to check for it, no? It would simply be a no-op.

But I suppose that only works well for actions and not selectors.

Reply:
We wouldn't have to check for it if we use the empty function however since they are supposed to return a promise, I was thinking of places where we would need to use `.then()` the code would probably look like this.
```js
    const request = getStoryById();
    if ( request ) {
      request.then( ( post ) => {

      } );
    }
```

and places where it's directly returning the API call like 
https://github.com/google/web-stories-wp/blob/b7412eaad26ad4cf43198806aa1d8dcca7c0588d/packages/story-editor/src/app/story/actions/useAutoSave.js#L45-L55

It would still be an inconsistent return and would need handling.

But by looking at the overall required changes, I think it would need fewer changes to use an empty function. Let me try.

Reply:
When I look more closely, it's not working very well. For example, here how would I do an early return if I don't want to call these if `getMedia` was not provided.
```js
fetchMediaStart({ pageToken: p }); 
const trackTiming = getTimeTracker('load_media'); 
```

https://github.com/google/web-stories-wp/blob/b7412eaad26ad4cf43198806aa1d8dcca7c0588d/packages/story-editor/src/app/media/local/useContextValueProvider.js#L82-L95

Reply:
Since the condition is required to be added in a lot of files, I would do it later in a separate PR after this is merged.

Reply:
That's a good point, thanks for exploring & explaining.

Let's keep it as-is then for now 👍